presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -816,9 +763,9 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
                     .run("grantTablePrivileges", stats.getGrantTablePrivileges().wrap(() -> {
                         try (HiveMetastoreClient metastoreClient = clientProvider.createMetastoreClient()) {
                             HivePrincipal hivePrincipal = toHivePrincipal(grantee);
-                            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(hivePrincipal, databaseName, tableName);
+                            Set<HivePrivilegeInfo> existingPrivileges = listTablePrivileges(databaseName, tableName, new PrestoPrincipal(USER, grantee));
```

Comment:
I've taken care of this in `Prepare HiveMetastore interface to accept ROLE for GRANT/REVOKE` commit